src/alchemlyb/workflows/abfe.py

```diff
@@ -0,0 +1,723 @@
+        self.logger = logging.getLogger('alchemlyb.workflows.ABFE')
+        self.logger.info('Initialise Alchemlyb ABFE Workflow')
+        self.logger.info(f'Alchemlyb Version: f{__version__}')
+        self.logger.info(f'Set Temperature to {T} K.')
+        self.logger.info(f'Set Software to {software}.')
+
+        self.update_units(units)
+
+        self.logger.info(f'Finding files with prefix: {prefix}, suffix: '
+                         f'{suffix} under directory {dir} produced by '
+                         f'{software}')
+        self.file_list = glob(dir + '/**/' + prefix + '*' + suffix,
+                         recursive=True)
+
+        self.logger.info(f'Found {len(self.file_list)} xvg files.')
+        self.logger.info("Unsorted file list: \n %s", '\n'.join(
+            self.file_list))
+
+        if software == 'GROMACS':
+            self.logger.info(f'Using {software} parser to read the data.')
+            self._extract_u_nk = gmx.extract_u_nk
+            self._extract_dHdl = gmx.extract_dHdl
+        elif software == 'AMBER':
+            self._extract_u_nk = amber.extract_u_nk
+            self._extract_dHdl = amber.extract_dHdl
+        else:
+            raise NotImplementedError(f'{software} parser not found.')
+
+    def read(self):
+        '''Read the u_nk and dHdL data from the
+        :attr:`~alchemlyb.workflows.ABFE.file_list`
+
+        Attributes
+        ----------
+        u_nk_list : list
+            A list of :class:`pandas.DataFrame` of u_nk.
+        dHdl_list : list
+            A list of :class:`pandas.DataFrame` of dHdl.
+        '''
+        u_nk_list = []
+        dHdl_list = []
+        for file in self.file_list:
+            try:
+                u_nk = self._extract_u_nk(file, T=self.T)
+                self.logger.info(
+                    f'Reading {len(u_nk)} lines of u_nk from {file}')
+                u_nk_list.append(u_nk)
+            except Exception as exc:
+                msg = f'Error reading read u_nk from {file}.'
+                if self.ignore_warnings:
+                    self.logger.exception(msg + f'\n{exc}\n' +
+                        'This exception is being ignored because ignore_warnings=True.')
+                else:
+                    self.logger.error(msg)
+                    raise exc from ValueError(msg)
+
+            try:
+                dhdl = self._extract_dHdl(file, T=self.T)
+                self.logger.info(
+                    f'Reading {len(dhdl)} lines of dhdl from {file}')
+                dHdl_list.append(dhdl)
+            except Exception as exc:
+                msg = f'Error reading read dhdl from {file}.'
+                if self.ignore_warnings:
+                    self.logger.exception(msg + f'\n{exc}\n' +
+                        'This exception is being ignored because ignore_warnings=True.')
+                else:
+                    self.logger.error(msg)
+                    raise exc from ValueError(msg)
```

Comment:
As above
```suggestion
                    raise OSError(msg) from exc
```

(Btw, `IOError` is just `OSError`... so we don't gain anything by using it, it's just more confusing.)

Reply:
Is this exception tested anywhere??